### DISCUSSION TITLE
32bit version

### DISCUSSION THREAD

**Discussion by WacKEDmaN** [OpenHantek/OpenHantek6022#383] [Q&A] [answered]
is it possible to get 32bit build please?
id like to run openhantek6022 on an old laptop that only has 32bit cpu... 
the official software works on 32bit...but as we all know its crap!...

**Comment by Ho-Ro** [comment, marked as answer]
I suspect you're using Windows, there are two hints:
- you mention the official SW
- you do not mention your OS, typical Windows user behaviour :)

If you need 32 bit support you have to build it by your own. The GitHub runner that builds the release packages supports only 64 bit since some years, release [3.1.5](https://github.com/OpenHantek/OpenHantek6022/releases/tag/3.1.5) was the latest 32bit build.
But you can have a look how it was compiled and try the same commands on your system - good luck.
Maybe also a kind Windows expert can provide a binary.

**Reply by WacKEDmaN** [reply]
thanks...yeah sorry ..typical windows user here!
im about to install linux on the laptop...so i should be right to build a 32bit on there...

**Reply by WacKEDmaN** [reply]
just to follow up.. i tried 32bit of version 3.1.5 on windows10 and windows7.. but no graph shown.. just what looked like error codes...  
so..installed debian 12 i386...and compiled from source....worked perfect first go!

**Reply by Ho-Ro** [reply]
Congrats!
I learned from others that Windows becomes difficult as soon you leave the standard applications - they have always issues with libusb programs that work flawlessly on all kind of Unixoids (Linux, *BSD, MacOS).
Another field of pain are the STM32 based HF tools, e.g. [tinySA](http://tinysa.org) and [NanoVNA](http://nanovna.com/), they always give headache when it comes to connecting these devices to the Win PC, half of their forum traffic is about "how do I access the USB based serial port?" instead of discussing the real use cases of these little gems.

**Reply by WacKEDmaN** [reply]
tbh ive never had issues with libusb on windows... i use RTL-SDR dongles and they have been fine.. unless M$ does some update that breaks the drivers!..usually just reinstalling the libusb driver would be enough to fix the problem....
i did manage to get sigrok/pulseview working on 32bit windows with the libusb driver... but i didnt like the app (as it was designed as a logic analyzer)...but it did work fine...

**Reply by Ho-Ro** [reply]
Ok, you're used to programming and electrical engineering. The mentioned tools are also used by hams (radio amateurs), many of them have a limited understanding of computer science. Even the use of zadig "sounds greek" to them.
I like pulseview, but only in combination with LAs :)
